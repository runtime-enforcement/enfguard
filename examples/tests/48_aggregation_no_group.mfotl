ALWAYS (FORALL s. s <- MAX(x; ; ONCE (C(x, x))) IMPLIES C(s, s))