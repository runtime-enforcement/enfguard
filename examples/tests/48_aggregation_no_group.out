The formula
 □[0s,∞) (∀s:int. (s:int <- MAX(x:int; ; ⧫[0s,∞) C(x:int, x:int))) → C(s:int, s:int))
is enforceable and types to
 □[0s,∞) ((∀s:int. ((s:int <- MAX(x:int; ; ⧫[0s,∞) C(x:int, x:int))) →:R (C(s:int, s:int) : Cau) : Cau)) : Cau) : Cau
[Enforcer] @1 reactively commands:
Cause:
C(-4611686018427387904, -4611686018427387904)
OK.
[Enforcer] @1 nothing to do proactively.
[Enforcer] @2 reactively commands:
Cause:
C(-4611686018427387904, -4611686018427387904)
OK.
[Enforcer] @2 nothing to do proactively.
[Enforcer] @3 reactively commands:
Cause:
C(-4611686018427387904, -4611686018427387904)
OK.
